R/radae.R

```diff
@@ -147,32 +148,27 @@ radae <- function(ADSL, # nolint
   # merge ADSL to be able to add AE date and study day variables
   ADAE <- dplyr::inner_join(ADAE, ADSL, by = c("STUDYID", "USUBJID")) %>% # nolint
     dplyr::rowwise() %>%
-    dplyr::mutate(trtsdt_int = as.numeric(as.Date(.data$TRTSDTM))) %>%
-    dplyr::mutate(trtedt_int = dplyr::case_when(
-      !is.na(.data$TRTEDTM) ~ as.numeric(as.Date(.data$TRTEDTM)),
-      is.na(.data$TRTEDTM) ~ floor(trtsdt_int + (study_duration_secs) / 86400)
-    )) %>%
-    dplyr::mutate(ASTDTM = as.POSIXct(
-      (sample(.data$trtsdt_int:.data$trtedt_int, size = 1) * 86400),
-      origin = "1970-01-01"
+    dplyr::mutate(TRTENDT = lubridate::date(dplyr::case_when(
+      is.na(.data$TRTEDTM) ~ lubridate::floor_date(lubridate::date(TRTSDTM) + study_duration_secs, unit = "day"),
+      TRUE ~ .data$TRTEDTM
+    ))) %>%
+    dplyr::mutate(ASTDTM = sample(
+      seq(lubridate::as_datetime(TRTSDTM), lubridate::as_datetime(TRTENDT), by = "day"),
+      size = 1
     )) %>%
-    dplyr::mutate(astdt_int = as.numeric(as.Date(.data$ASTDTM))) %>%
-    dplyr::mutate(ASTDY = ceiling(as.numeric(difftime(.data$ASTDTM, .data$TRTSDTM, units = "days")))) %>%
+    dplyr::mutate(ASTDY = ceiling(difftime(.data$ASTDTM, .data$TRTSDTM, units = "days"))) %>%
```

Comment:
I was thinking about this one, as it still uses the base R way. I run a few experiments here following the following:
```r
library(dplyr)
library(lubridate)
library(microbenchmark)

today <- as_datetime("2020-11-04 04:08:58 UTC")
dob <- as_datetime("2021-04-15 04:08:58 UTC")

microbenchmark(
  interval(dob, today) / days(1),
  as.duration(interval(dob,today)) %/% as.duration(days(1)),
  interval(dob, today) %>% as.numeric('days'),
  dob %--% today / ddays(1),
  difftime(dob, today, units = "days"),
  as.numeric(today - dob)
)
```
The first thing to notice is that in the diffdate has a formatted print which is going away only in the mutate (I guess you took out as.numeric for that reason). 
```r
>   as.numeric(today - dob)
[1] -162
> interval(dob, today) / days(1)
[1] -162
> as.duration(interval(dob,today)) %/% as.duration(days(1))
[1] -162
> interval(dob, today) %>% as.numeric('days')
[1] -162
> dob %--% today / ddays(1)
[1] -162
> difftime(dob, today, units = "days")
Time difference of 162 days
```
and here it is the clear winner for today's speed contest:
```
Unit: microseconds
                                                     expr    min      lq     mean  median      uq     max neval
                             interval(dob, today)/days(1) 2093.6 2384.05 2957.089 2600.80 2858.45 17310.0   100
 as.duration(interval(dob, today))%/%as.duration(days(1)) 1109.4 1321.35 1561.055 1390.25 1532.10 15065.7   100
              interval(dob, today) %>% as.numeric("days")  676.2  786.55 1011.419  835.40  898.75 16599.0   100
                                  dob %--% today/ddays(1)  588.9  698.25  782.175  775.05  820.45  1177.9   100
                     difftime(dob, today, units = "days")   13.2   22.10   34.145   31.20   41.00    86.5   100
                                  as.numeric(today - dob)   19.5   27.30   42.064   41.65   52.40    91.1   100
```
Question: do you think we should use the fastest option or start using lubridate always, also for diffs?

Reply:
I think it is fine as you did. Very clear and fast.